### PR TITLE
dashboard report menu to preserve state and scroll

### PR DESCRIPTION
**Pull request by djvoa12**
**Summary:**
Dashboard report list now keeps its report-setting state via query params and also preserves its scroll.
- Report ID - `/dashboard?report=1`
- Report Year - `/dashboard?reportYear=2016`

**UI:**
Screen Shot 2019-05-14 at 11 24 44 AM

Fixes https://github.com/ilios/frontend/issues/4541

### CODE REVIEW

File: app/components/dashboard-myreports.js [hunk 1]
```diff
@@ -45,34 +77,34 @@ export default Component.extend(ReportTitleMixin, {
       });
     });
   }),
-  reportResultsList: computed('selectedReport', 'selectedYear', function(){
-    const report = this.selectedReport;
+
+  reportResultsList: computed('selectedReport', 'selectedYear', async function() {
+    const report = await this.selectedReport;
```
Comment (jrjohnson) [inline app/components/dashboard-myreports.js]: `selectedReport` looks like a value and not a promise, any reason for the `await` here?

File: tests/integration/components/myreports-list-item-test.js [hunk 2]
```diff
@@ -14,13 +14,15 @@ module('Integration | Component | myreports list item', function(hooks) {
       title: 'Lorem Ipsum'
     });
 
-    const reportModel = await this.owner.lookup('service:store').find('report', report.id);
+    const reportModel = await this.owner
```
Comment (jrjohnson) [inline tests/integration/components/myreports-list-item-test.js]: This should remain a single line, multi-line is unnecessary here and makes it harder to see where the promise is that is being `await`ed.

**Comment by jrjohnson** [comment]
This looks great! Preserving the scroll state here seems particularly useful. Couple of minor tweaks in code, otherwise looks ready if @dartajax gives it a 👍

File: app/components/dashboard-myreports.js (hunk 1 above)
Comment (djvoa12) [reply app/components/dashboard-myreports.js]: Probably an error.

File: tests/integration/components/myreports-list-item-test.js (hunk 2 above)
Comment (djvoa12) [reply tests/integration/components/myreports-list-item-test.js]: Ok

**Pull request merged**